# RonBot

https://en.wikipedia.org/wiki/User:RonBot

Bot is no longer active, operator has since passed away

tried to look for discussion of malfunctionmost recent talk on user's page is from 2020 and can't find anything there

discussion about brokenimage task possibly being broken in March 2019
https://en.wikipedia.org/wiki/Wikipedia:Bots/Noticeboard/Archive_13#User:RonBot_trouble_possibly_in_need_of_intervention

discussion about footballer category task in July 2018 https://en.wikipedia.org/wiki/Wikipedia:Bot_requests/Archive_76#Association_footballers_not_categorized_by_position


In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
from datetime import datetime

In [4]:
df = pd.read_csv('revs_scored_jan.tsv', sep='\t', header=0)

In [5]:
days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [6]:
sdf_ronbot = df[df.day<22]
sdf_ronbot = sdf_ronbot[sdf_ronbot.user_text == "RonBot"]

sdf_reverted = sdf_ronbot[sdf_ronbot.is_reverted] 
sdf_reverted = sdf_reverted[sdf_reverted.seconds_to_revert.astype('str').astype('int')<86400]

In [7]:
sdf1 = sdf_reverted.loc[sdf_reverted.is_reverted, ["rev_id", "user_text", "revert_id", "page_namespace"]]
sdf1.revert_id = sdf1.revert_id.astype('int')

sdf2 = df[df.user_is_bot == False]
sdf2 = sdf2[sdf2.is_revert]
sdf2 = sdf2[["rev_id", 'user_text']]

In [8]:
reverts_by_human = pd.merge(sdf1, sdf2, 
             how='inner', 
             left_on='revert_id', 
             right_on='rev_id', 
             suffixes=('', '_reverter')
)

In [9]:
reverts_by_human0 = reverts_by_human[reverts_by_human.page_namespace == 0]

In [10]:
# summary of variables
#   sdf_ronbot = all edits by ronbot w/in time frame
#   sdf_reverted = all reverted edits by ronbot w/in time frame
#   reverts_by_human = all edits of Ronbot that were reverted by a human
#   reverts_by_human0 = all edits by Ronbot in namespace 0 reverted by a human

## sample of edits reverted by humans in namespace 0

In [14]:
len(reverts_by_human0)

429

In [15]:
reverts_by_human0.sample(n=20, random_state=1).reset_index()

index     rev_id user_text  revert_id  page_namespace  rev_id_reverter  \
0     346  877208766    RonBot  877271049               0        877271049   
1     327  877208664    RonBot  877276144               0        877276144   
2     283  877208212    RonBot  877208417               0        877208417   
3     396  877209086    RonBot  877269769               0        877269769   
4     181  877202703    RonBot  877218672               0        877218672   
5     291  877208346    RonBot  877277625               0        877277625   
6     447  877277070    RonBot  877295106               0        877295106   
7       6  876292953    RonBot  876296168               0        876296168   
8     361  877208884    RonBot  877270435               0        877270435   
9     505  877279563    RonBot  877281409               0        877281409   
10    344  877208746    RonBot  877273045               0        877273045   
11    462  877277582    RonBot  877332141               0        877332141   
12      7  876293084    RonBot  876294988               0        876294988   
13    682  879469682    RonBot  879469952               0        879469952   
14    298  877208422    RonBot  877218437               0        877218437   
15    355  877208840    RonBot  877270717               0        877270717   
16    403  877209113    RonBot  877211709               0        877211709   
17    337  877208707    RonBot  877275779               0        877275779   
18    136  877201219    RonBot  877328579               0        877328579   
19    241  877207378    RonBot  877290483               0        877290483   

      user_text_reverter  
0           Inter&anthro  
1           Inter&anthro  
2           Inter&anthro  
3           Inter&anthro  
4           GiantSnowman  
5           Inter&anthro  
6        Jmorrison230582  
7                 Vsmith  
8           Inter&anthro  
9         Walter Görlitz  
10          Inter&anthro  
11         Mattythewhite  
12              Gotitbro  
13           KylieTastic  
14          GiantSnowman  
15          Inter&anthro  
16          Inter&anthro  
17          Inter&anthro  
18         Mattythewhite  
19  Daemonickangaroo2018

0. 877208766 on Adil Mezgour page, for task 7. ronbot adds category (footballers not categorized by position). user inter%anthro undoes it, says position category already present. indeed, article does list category "association football forwards" already
1. 877208664 on Adham El Idrissi page, for task 7. ronbot adds category. user inter&anthro deletes it, says position category already present. indeed, category "association football forwards" is already present.
2. 877208212 on Adamo Coulibaly page, for task 7. ronbot adds category (footballers not categorized by position). user Inter%anthro undoes it, says position category already present. indeed, article does list category "association football forwards" already
3. 877209086 on dif page, for task 7. ronbot adds category (footballers not categorized by position). user Inter%anthro undoes it, says position category already present. indeed, article does list category "association football midfielders" already
4. 877202703 same as above, user GiantSnowman reverts, in midfielders category
5. 877208346 same as above, user inter&anthro reverts, in defenders category, edits are 12 hours apart on Jan 7
6. 877277070 same as above, user Jmorrison230582 reverts, in category forwards, edits are 2 hours apart on Jan 7
7. 876292953 on Eastern world page, RonBot adds broken image template. another user vsmith reverts to edit before alleged vandalism that broke the image ***not a malfunction***. edits about 30 mins apart, on Jan 1.
8. 877208884 same footballer edits, reverting user is Inter&anthro, forwards category already present, edits about 12 hours apart on Jan 7
9. 877279563 same footballer edits, reverting user is walter gorlitz, in category defenders, about 10 mins apart 7 Jan
10. 877208746 same footballer edits, reverting user is Inter&anthro, in category defenders, edits about 11 hours apart on Jan 7
11. 877277582 same footballer edits, reverting user Mattythewhite, in category midfielders,  edits about 7 hours apart on Jan 7
12. 876293084 on Johnny Sins page, RonBot adds a BrokenImage template. user Gotitbro reverts, says no broken image. indeed the image seems to be correct.
13. 879469682 on Daksha page, RonBot adds BrokenImage template (task 12). user KylieTastic reverts to a different edit to fix the name of the image. ***not a malfunction***
14. 877208422 same football edits, reverting user is GiantSnowman, in category midfielders, edits are about 3 hours apart Jan 7
15. 877208840 same football edits, reverting user is inter&anthro, in category defenders, 12 hours apart on Jan 7
16. 877209113 same football edits, reverting user is Inter&anthro, in category defenders, edits about 1 hour apart on Jan 7
17. 877208707 same football edits, reverting user Inter&anthro, in category forwards, edits about 11 hours apart on Jan 7
18. 877201219 same football edits, reverting user Mattythewhite, in category forwards, edits about 20 hours apart. first edit is 30 mins before midnight on 6 Jan, second 18:00 on Jan 7
19. 877207378 same football edits, reverting user Daemonickangaroo2018, in category midfielders,  edits about 13 hours apart on Jan 7